### PR TITLE
Stress test flexible config

### PR DESCRIPTION
**Pull request by sreecha**
Note: This pull request subsumes https://github.com/grpc/grpc/pull/5922

- Run stress tests now accepts a config file instead of command line args. 
  -- It would now be possible to run stress tests in mixed configurations (for example, a tsan client talking to an asan server) or multiple instances of client-server configs in the same GKE cluster..

  -- Since this was a major rewrite of the file `run_stress_tests_on_gke.py`,  I felt it is better to just delete it and create a new file `run_on_gke.py`

  -- The two files under the `configs` directory are just examples. The file `opt.json` has config for just opt clients against opt servers while the other file `opt-tsan-asan.json` has config that creates three groups of clients (`opt`, `asan` and `tsan`) each running against its corresponding server (`opt`, `asan` and `tsan`) all in the same GKE cluster at the same time. 

 -- We can add more files as we think of more test configurations

- This includes Vijay's changes to build clang
- I also used this opportunity to generate Dockerfile from templates

### CODE REVIEW

File: tools/run_tests/stress_test/configs/opt-tsan-asan.json [hunk 1]
```diff
@@ -0,0 +1,135 @@
+{
```
Comment (jtattermusch) [inline tools/run_tests/stress_test/configs/opt-tsan-asan.json]: I think JSON config is fine (at least for now), but please note that we've been kind of pushing .yaml configurations within our repo.

**Comment by jtattermusch** [comment]
Interestingly enough, objective C tests on travis didn't run for this PR :-). Looking at the changes, I see no way this PR could influence objective C and all other tests are green, so I will merge.

**Comment by jtattermusch** [comment]
LGTM.

**Comment by sreecha** [comment]
Thanks for reviewing it!

**Pull request merged**